### ISSUE TITLE
grid lines, field borders, etc.

### ISSUE THREAD

**Issue by shawna-slh** [w3c/low-vision-a11y-tf#54]
​from: https://lists.w3.org/Archives/Public/public-low-vision-comments/2016JanMar/0005.html

One particular thing that I found missing is the ability to customise grids and leading lines. In our tests with low vision users, we often had the issue that light grey grids weren't perceived and items not allocated accordingly.Say, in a Samsung calendar grid, the salient info can easily be allocated to the wrong day (see Fig 10 in http://www.incobs.de/tests/items/tablet-tests-mit-sehbehinderten-nutzern-gebrauchstauglichkeit.html)

A similar issue is controls that have a large offset from the label. _[ed note: this is covered in [3.6.2 Proximity of Related Information](http://w3c.github.io/low-vision-a11y-tf/requirements.html#proximity-of-related-information)]_ Moving the visible section (e.g. on a mobile device) to bring related control or label in view is much harder if there is no visible guiding line helping in making the link (e.g. (on Windows Mobile). Web  / App developers may be required to ensure 4.5:1  contrast of leading lines / grids where these aid perception/navigation (this could presumably also be a system-level setting where contrast of grids and dividing / leading lines can be increased).

**Comment by shawna-slh** [comment]
A similar issue is field borders not being visible enough.

"JDA: Contrast on borders for form fields. Some form fields don't have borders at all. Example gofundme.com I've heard from designers that they need clear guidance on ratios for border contrast or they won't implement. Having a best practice or hearing on that users can adjust the border color means they will NOT take action." (from [open issues](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/LVTF_Open_Issues))

Does this fit as an example or expansion of existing user need (instead of adding a new one)? It relates not only to contrast, but also to thickness.

Is the user need that people can see borders, girds, lines, and such? Can we leave it at that and not need to say that users can customize those?

**Comment by lauracarlson** [comment]
I agree with Detlev Fischer in his comment regarding problems with light grey not being perceived. 

I also agree with JDA that clear guidance on ratios for border contrast is what designers need. Without that, designers will continue using light grey on white. Perhaps set a baseline and allow customization from there. 

Detlev suggested 4.5:1 contrast of leading lines / grids where these aid perception/navigation. As he said this does not rule out a system-level setting where contrast of grids and dividing / leading lines can be increased).

**Comment by allanj-uaag** [comment]
It sounds like the 4.5:1 should apply to (hesitate to say "all") elements, such as: 
https://www.w3.org/TR/html-markup/elements-by-function.html
section elements
grouping elements
text level semantics
edit elements
form elements
CSS (border, grids, what else?)

Is there a minimum size or width (thickness) for borders, grids, bullets, etc. ?

**Comment by allanj-uaag** [comment]
see [action-51](http://www.w3.org/WAI/GL/low-vision-a11y-tf/track/actions/51)

**Comment by shawna-slh** [comment]
[e-mail thread](https://lists.w3.org/Archives/Public/public-low-vision-a11y-tf/2016Apr/0027.html)

**Comment by allanj-uaag** [comment]
see https://lists.w3.org/Archives/Public/public-low-vision-a11y-tf/2016Apr/0040.html
This seems wordy. 
To me, this belongs in 3.3 Perceiving - Perceiving includes being able to recognize individual letters based on their characteristics, and see non-text information and interface elements. 
There is the author side of things - they need to style things when authoring so users can see them. 
There is a browser responsibility - default rendering of elements must meet WCAG 1.4.3
This issue is also closely related to 3.5.1 Element Level Customization - though this seems more of a solution - user settings in a UA).

Propose: 
Need: Users need to be able to see  non-text information and interface elements -- e.g. form control outlines (borders), grid lines (table cell borders), horizontal rules, leading lines (dots), highlights (select box options).

Explanation: Often form fields, tables, and special blocks of text -- like definition or navigation sections, are called out using borders or changes in background color. In these cases, the borders and altered background colors should provide sufficient contrast with the all adjacent regions of the page. This includes all borders used for form fields, radio buttons, check boxes and focus indicators. Objects that are distinguished by alternative colors including form fields should also have sufficient contrast as defined in the WCAG 1.4.3.

> from Wayne - Need: When the content author uses visual cues to guide the visual user’s
> scanning of the page, a user with reduced contrast sensitivity should be
> able to perceive the visual cues being used. Specifically, all such visual
> cues must provide sufficient contrast to support detection by users with
> reduced contrast sensitivity.
> 
> Explanation: Often form fields and special blocks of text like definition
> or navigation sections are called out using borders or changes in
> background color. In these cases, the borders and altered background colors
> should provide sufficient contrast with the all adjacent regions of the
> page. This includes all borders used for form fields, radio buttons, check
> boxes and focus indicators. Objects that are distinguished by alternative
> colors including form fields should also have sufficient contrast as
> defined in the WCAG 2.0 glossary.

**Comment by ghost** [comment]
agree with allanj-uaag Propose. Form controls / elements with unclear borders appear simply as blank space, very common in high contrast / invert colors

**Comment by shawna-slh** [comment]
also other color contrast issues -- see SC discussion from [15 Sept 2016](https://www.w3.org/2016/09/15-lvtf-minutes.html)

**Comment by shawna-slh** [comment]
Specific requirements are being defined in WCAG 2.1. For this User Needs/Requirements doc, we want to explain the issue, and do not need to be prescriptive, e.g., with minimum contrast ratio. The goal for this doc is for people to understand the user experience and the user need. We also want to keep this as simple as feasible, e.g., can we cover UI controls and graphics together in one section? UI controals are a kind of graphical element. :-)

Straw proposal: After [3.1.2 Text Contrast](https://www.w3.org/TR/low-vision-needs/#text-contrast) add a new requirement:

> **3.1.3 Graphics Contrast**
If there is not sufficient contrast with the background, some users cannot see:
•	User interface controls, such as text field borders, buttons, and visual focus indicators
•	Table cell borders, gridlines, and other dividers 
•	Information in graphs, photos, and other images
User interface controls and important information in graphics should have sufficient contrast and thickness. Ideally, users can also change contrast in user interface controls, per 3.5.1 Element-level Customization.
**User Need – Graphics Contrast:** User interface controls and graphical elements have sufficient contrast and thickness.

**Comment by allanj-uaag** [comment]
(fyi: e-mail to commenter: https://lists.w3.org/Archives/Public/public-low-vision-comments/2017AprJun/0002.html )

**Comment by lauracarlson** [comment]
Hi @slhenry  ,

I previously "Thumbs Up" [your proposal](https://github.com/w3c/low-vision-a11y-tf/issues/54#issuecomment-294968939). Seems fine to me.
